### DISCUSSION TITLE
Interfaces vy to solidity

### DISCUSSION THREAD

**Discussion by kaleb-keny** [vyperlang/vyper#2271] [General]
I have a question  please on setting up an interface to a contract on written in solidity. So the contract i'm referring to has the the eblow solc code, on this address `0x915D1c9dF12142B535F6a7437F0196D80bCCC1BD`
The issue with it, as you can it takes in as an argument an array or list without pre-defined range. The issue i'm facing when building a vy interface for it is that it's not possible from what I read to make dynamic arrays or lists. So the interface I implemented down doesn't seem to work maybe. is there like a work around possible, any example is appreciated 🙏🏼 ?

```
    function expireMarkets(address[] calldata markets) external notPaused {
        for (uint i = 0; i < markets.length; i++) {
            address market = markets[i];

            // The market itself handles decrementing the total deposits.
            BinaryOptionMarket(market).expire(msg.sender);
            // Note that we required that the market is known, which guarantees
            // its index is defined and that the list of markets is not empty.
            _maturedMarkets.remove(market);
            emit MarketExpired(market);
        }
    }
```
Inteface:
```
####version 

interface optionManager:
    def expireMarkets(addressList:address[1]): nonpayable

##Init Here

@external
def execute(_contract: address, _target: address[1]):
    assert msg.sender == self.owner, "caller not owner"
    optionManager(_contract).expireMarkets(_target)
```

**Comment by fubuloubu** [comment]
You can try to work around it manually using [`raw_call`](https://vyper.readthedocs.io/en/stable/built-in-functions.html?highlight=raw_call#raw_call), until we add dynamic array support w/ https://github.com/vyperlang/vyper/issues/1440

Would look something like:
```python
raw_call(
    _contract,
    concat(
         method_id("expireMarkets(address[])"),  # method id (4 bytes)
         68,  # offset to dynamic array in ABI-encoded data (in bytes)
         32,  # length of dynamic array (in bytes)
         convert(_target, bytes32),  # array (length 1)
    ),
)
```

**Reply by fubuloubu** [reply]
Futz with the numbers until it works

**Reply by kaleb-keny** [reply]
appreciate it, I will try raw calls, thanks! I didn't know that was really the problem until you confirmed it, I was just speculating 😅

**Reply by fubuloubu** [reply]
Yeah, it's been a feature long time in the making. Add dynamic array support is pretty complicated, and a source of a ton of bugs in Solidity tbh. We want to make sure we get it right.

**Reply by kaleb-keny** [reply]
Yeah honestly, I know the feeling, dynamic are pretty hard in VB, but in python everything is kind of dynamic, so I just assumed it would inherit it....

**Reply by fubuloubu** [reply]
Haha, no I wish. Vyper may be Pythonic, but the code generation process is pretty specific to the EVM, more like C/C++

**Reply by kaleb-keny** [reply]
@fubuloubu worked perfectly as you described... Thanks a lot! appreciate it... is there someone building out like a eli5 repo that gives examples on how to interact with some contracts? like this is how you setup a a flash loan with vy, this is how you call raw transactions...  if yes, i'd love to contribute gitcoin-wize

**Reply by fubuloubu** [reply]
I mean really, we just need to add support for dynamic arrays into the language lol

That issue is here: https://github.com/vyperlang/vyper/issues/1440

**Reply by kaleb-keny** [reply]
you over-estimate our normie brains lol in writing contracts 😄  
What I meant, maybe something that would help gain more traction of vy, and this is just my opinion, maybe others disagree, idk, are like example repositories of interactions with defi contracts... 
the docs are great don't get me wrong, they are actually fantastic 😄  but for example for a certain use case like flash loan or like burning chi  to save gas, or like simply transferring an erc-20 in a contract lol (approve then transfer with interface)... 
there are like plenty of repos in solc with this stuff ....  but for vy, maybe it's just me who didn't manage to find them yet, I try to look in banteg's repo sometimes 😊 .... but they are a bit harder to come by... 
but please don't misunderstand what i'm saying, what you guys are building is amazing honestly 🙏🏼 , and vy just makes sense when I read it, as opposed to solc....

**Reply by fubuloubu** [reply]
ah sorry, I thought you meant for this Uniswap interface specifically. This is a stopgap fix until we add the ability to define dynamic arrays, which you could just do like:

```python
interface OptionManager:
    def expireMarkets(addressList: Array[address, 3]): nonpayable


@external
def execute():
    assert msg.sender == self.owner, "caller not owner"
    OptionManager(self.option).expireMarkets([self.target1, self.target2])
```
